### PR TITLE
Brew: Add greedy preference; upgrade and doctor commands.

### PR DESCRIPTION
**Pull request by nhojb**
## Description

- Add "greedy" upgrade preference.
- Add upgrade and doctor commands.

## Type of change

- Bug fix / improvement for my extension

## Screencast

Screenshot 2021-12-21 at 10 29 40
Screenshot 2021-12-21 at 10 27 46
Screenshot 2021-12-21 at 10 27 51

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/brew/src/brew.ts [hunk 1]
```diff
@@ -99,6 +105,34 @@ export function brewPath(suffix: string): string {
...
 
+/// Commands
+
+export async function brewDoctorCommand(): Promise<string> {
+  try {
+    const output = await execp(`${brewExecutable} doctor`);
+    return output.stdout;
+  } catch (err) {
+    const execErr = err as ExecError;
+    if (execErr?.code == 1) {
```
Comment (fe9lix) [inline extensions/brew/src/brew.ts:116]: Probably you want strict equality here? (===)

**Review by fe9lix — COMMENTED** [review]
Review done. @nhojb One issue that I had when running brew doctor: It reports "Your system is ready to brew" when I run it in the terminal but shows some errors when running it through the extension command. I guess that's because the PATH variable is not set?

**Comment by nhojb** [comment]
> Review done. @nhojb One issue that I had when running brew doctor: It reports "Your system is ready to brew" when I run it in the terminal but shows some errors when running it through the extension command. I guess that's because the PATH variable is not set?

Yes, I get the same problem. I think `launchd` (and hence Raycast) pick up a different environment than a shell you might run from your Terminal. I almost didn't include the `doctor` command for this reason. Perhaps it's better just to remove it (I originally wrote it to try and debug issues with the brew process running in the extension process)?

**Comment by nhojb** [comment]
On second thoughts I think I'll just remove the `doctor` command.

**Comment by fe9lix** [comment]
Yeah, the env is different since it's a different process, launched outside of the shell – the shell usually picks the customized PATH up from one of the known config files (~.zshrc, etc.) so brew doctor would yield different output when launched from there.

**Review by fe9lix — APPROVED** [review]

**Pull request merged**